### PR TITLE
[MS-491] limit fingerprint scan attempts

### PR DESCRIPTION
**Pull request by alexandr-simprints**
There is now a limit to the amount of 'No Finger detected' scans before the user is automatically navigated to the next finger.

- The default value is `40` and is stored in the `IsNoFingerDetectedLimitReachedUseCase.MAXIMUM_LIMIT_OF_NO_FINGER_DETECTED_SCANS`
- The value is set in the Vulcan under `Configuration -> Fingerprint -> SimMatcher`
- If the value is not in the [2..40] range, the default value of `40` is used

Other Changes
To accommodate the newly added `noFingerDetectedThreshold`, the `CaptureState` sealed class was refactored to have the following subclasses:
- `CaptureState.NotCollected`
- `CaptureState.Skipped`
- `CaptureState.ScanProcess` - number of bad scans and no-finger-detected is stored here

### CODE REVIEW

File: fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/usecase/IsNoFingerDetectedLimitReachedUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+internal class IsNoFingerDetectedLimitReachedUseCase @Inject constructor() {
+    operator fun invoke(
+        fingerState: CaptureState,
+        configuration: FingerprintConfiguration
+
+    ): Boolean = when (fingerState) {
+        is CaptureState.ScanProcess -> {
+            val noFingerDetectedThreshold =
+                configuration.secugenSimMatcher?.maxCaptureAttempts?.noFingerDetected?.takeIf { it > 1 }
```
Comment (BurningAXE) [inline fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/usecase/IsNoFingerDetectedLimitReachedUseCase.kt:15]: Is this feature only applicable to SimMatcher? If so, we should take care to either not call this use case when using NEC or pass the SDK used currently and skip the check for NEC.

**Review by BurningAXE — CHANGES_REQUESTED** [review]
We should handle NEC, too. Otherwise looks good!

File: fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/usecase/IsNoFingerDetectedLimitReachedUseCase.kt (hunk 1 above)
Comment (alexandr-simprints) [reply fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/usecase/IsNoFingerDetectedLimitReachedUseCase.kt:15]: That's a good question, actually. Let's take it to Slack first

**Review by luhmirin-s — APPROVED** [review]

File: fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/usecase/IsNoFingerDetectedLimitReachedUseCase.kt (hunk 1 above)
Comment (BurningAXE) [reply fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/usecase/IsNoFingerDetectedLimitReachedUseCase.kt:15]: OK, then the use case should lookup the configuration for the particular SDK used. Currently it's always checking the SimMatcher config.

Comment (alexandr-simprints) [reply fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/usecase/IsNoFingerDetectedLimitReachedUseCase.kt:15]: Great spot! Fixing it now

**Review by luhmirin-s — APPROVED** [review]

**Review by BurningAXE — APPROVED** [review]

**Pull request merged**